### ISSUE TITLE
Separate fails when File[] is given

### ISSUE THREAD

**Issue by jjkoehorst** [common-workflow-language/cwltool#1337]
## Expected Behavior
When using separate false I expect that arguments are not separated. This works for string, int, file but not for File[] when calling the workflow. 

## Workflow Code
```
#!/usr/bin/env cwl-runner
cwlVersion: v1.2
class: CommandLineTool

inputs:
  one:
    type: File[]
    inputBinding:
      prefix: 'one='
      separate: false
  two:
    type: string
    default: "2"
    inputBinding:
      prefix: 'two='
      separate: false

outputs:
  out:
    type: stdout
stdout: sep_stdout.txt

baseCommand: [echo]
```

Corresponding workflow

```
#!/usr/bin/env cwl-runner
cwlVersion: v1.2
class: Workflow

requirements:
   - class: StepInputExpressionRequirement

inputs:
  one:
    type: File[]

steps:
  separate:
    run: ./separate.cwl
    in: 
      one: one
    out: [out]

outputs:
  separate:
    type: File
    outputSource: separate/out
```

## Full Traceback
```pytb
INFO /usr/local/bin/cwltool 3.0.20200807132242
INFO Resolved 'sep_workflow.cwl' to 'file:///Users/bla/sep_workflow.cwl'
sep_workflow.cwl:13:3: object id `sep_workflow.cwl#separate` previously defined
INFO [workflow ] start
INFO [workflow ] starting step separate
INFO [step separate] start
INFO [job separate] /private/tmp/docker_tmpz8fq2oe_$ echo \
    one= \
    /private/var/folders/7s/1n66vsfj1c93hm09w2xz747c0000gn/T/tmpo6gw9fki/stg86055357-807a-4835-bfc5-bcab8d69392f/test.txt \
    two=2 > /private/tmp/docker_tmpz8fq2oe_/sep_stdout.txt
INFO [job separate] completed success
INFO [step separate] completed success
INFO [workflow ] completed success
{
    "separate": {
        "location": "file:///Users/bla/sep_stdout.txt",
        "basename": "sep_stdout.txt",
        "class": "File",
        "checksum": "sha1$1238c7933e251d901017ed8a030f2c4c6d03e612",
        "size": 129,
        "path": "/Users/bla/sep_stdout.txt"
    }
}
INFO Final process status is success```

There is now a space after `one=`

## Your Environment
* cwltool version: 3.0.20200807132242
Check using ``cwltool --version``

**Comment by mr-c** [comment]
Hello @jjkoehorst ! I just realized that we had this question on the forum recently, and I think this answer should work for you as well: https://cwl.discourse.group/t/cwl-pass-array-of-elements-with-repeated-input-flags/190/2

**Comment by jjkoehorst** [comment]
Hey @mr-c the problem is that when calling the cwl file directly it works but when calling it though a cwl workflow it does not work.

**Comment by jjkoehorst** [comment]
This seems to solve the issue but what is the difference with File[] vs below?

```
  one:
    type:
      - "null"
      - type: array
        items: File
        inputBinding:
          prefix: 'in='
          separate: false
```

**Comment by mr-c** [comment]
> This seems to solve the issue but what is the difference with File[] vs below?
> 
> ```
>   one:
>     type:
>       - "null"
>       - type: array
>         items: File
>         inputBinding:
>           prefix: 'in='
>           separate: false
> ```

This puts the prefix before each item of the array, your first example put the prefix before the entire array, only once.

Try testing with multiple input files.

However I suspect that this is a separate issue from the extra space that depends on directly running versions in a Workflow.
